### PR TITLE
Speed screen improvements

### PR DESCRIPTION
**Pull request by peterpoetzi**

### CODE REVIEW

File: main/displays/measurementdisplay.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+// 3rdparty lib includes
+#include <widgets/label.h>
+#include <widgets/progressbar.h>
+#include <espchrono.h>
+#include <optional>
+
+// local includes
+#include "bobbydisplaywithtitle.h"
+#include "modes/ignoreinputmode.h"
```
Comment (CommanderRedYT) [inline main/displays/measurementdisplay.h:11]: Why do you need ignore input mode for something that measures driving statistics?

File: main/displays/measurementdisplay.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        }
+        emplaceMenuItem(std::move(backButton));
+    }
+
+    //take measurement and add menu items
+    while (distance_at_start && avgSpeedKmh + configs.measurementMode.measurementInterval.value() > m_max_speed) {
+        m_max_speed += configs.measurementMode.measurementInterval.value();
+
+        Result result = {
+                .time = std::chrono::floor<std::chrono::milliseconds>(espchrono::millis_clock::now() - *m_start_time),
```
Comment (CommanderRedYT) [inline main/displays/measurementdisplay.cpp:129]: m_start_time could be std::nullopt

File: main/displays/measurementdisplay.cpp [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        void triggered() override {
+            //TODO: call clearMeasurements()
+        }
+    };
+
+
+
+    class MeasurementDisplayItem : public MenuItem {
+    public:
+        MeasurementDisplayItem(const MeasurementDisplay::Result result) : m_result{result} {}
```
Comment (CommanderRedYT) [inline main/displays/measurementdisplay.cpp:48]: Remove copy

File: main/displays/speedinfodisplay.cpp [hunk 4]
```diff
@@ -46,15 +46,16 @@ void SpeedInfoDisplay::redraw()
     if (const auto avgVoltage = controllers.getAvgVoltage(); avgVoltage)
     {
         auto watt = sumCurrent * *avgVoltage;
-        auto wh_per_km = std::abs(avgSpeedKmh) > 0.1 ? watt / std::abs(avgSpeedKmh) : 0;
 
         m_voltageLabel.redraw(fmt::format("{:.1f} V", avgVoltage.value()));
-        m_currentConsumptionLabel.redraw(fmt::format("{:.1f} Wh/km", wh_per_km));
+        tft.setTextSize(2);
```
Comment (CommanderRedYT) [inline main/displays/speedinfodisplay.cpp:51]: You could also use &s / &m as documented in 0xFEEDC0DE64/esp-gui-lib

File: main/newsettings.h [hunk 5]
```diff
@@ -438,6 +438,12 @@ class ConfigContainer
         ConfigWrapperLegacy<uint32_t> totalCentimeters  {0,                                      DoReset,   {},                         "totalCentimeter"     };
     } savedStatistics;
 
+    struct {
+        ConfigWrapperLegacy<uint8_t> measurementInterval{1,                                      DoReset,   {},                         "measurementiv"       };
+    } measurementMode;
+
```
Comment (CommanderRedYT) [inline main/newsettings.h:444]: Why so many empty lines?

**Review by CommanderRedYT — CHANGES_REQUESTED** [review]
Fix formatting

File: main/displays/measurementdisplay.h (hunk 1 above)
Comment (peterpoetzi) [reply main/displays/measurementdisplay.h]: Will be addressed in https://github.com/bobbycar-graz/bobbycar-boardcomputer-firmware/pull/352

File: main/displays/measurementdisplay.cpp (hunk 2 above)
Comment (peterpoetzi) [reply main/displays/measurementdisplay.cpp]: Will be addressed in https://github.com/bobbycar-graz/bobbycar-boardcomputer-firmware/pull/352

File: main/displays/measurementdisplay.cpp (hunk 3 above)
Comment (peterpoetzi) [reply main/displays/measurementdisplay.cpp]: Will be addressed in https://github.com/bobbycar-graz/bobbycar-boardcomputer-firmware/pull/352

File: main/displays/speedinfodisplay.cpp (hunk 4 above)
Comment (peterpoetzi) [reply main/displays/speedinfodisplay.cpp:51]: This does not work for setting the text size, only for changing the font. However, I do not like the large font and find the regular font with 2x the size more readable

File: main/newsettings.h (hunk 5 above)
Comment (peterpoetzi) [reply main/newsettings.h]: Will be addressed in https://github.com/bobbycar-graz/bobbycar-boardcomputer-firmware/pull/352

**Review by CommanderRedYT — APPROVED** [review]

**Pull request merged**